static/js/util/profile_edit.js

```diff
@@ -12,72 +15,141 @@ import moment from 'moment';
-    let clone = Object.assign({}, profile);
-    clone[key] = value ? value.value : null;
+    let clone = _.deepClone(profile);
+    _.set(clone, keySet, value? value.value : null)
     updateProfile(clone);
   };
   return <div>
     <Select
       options={options}
-      value={profile[key]}
+      value={_.get(profile, keySet, undefined)}
       placeholder={label}
       onChange={onChange} />
-    <span className="validation-error-text">{error}</span>
+    <span className="validation-error-text">{_.get(errors, keySet, undefined)}</span>
   </div>;
 }
 
 // bind this to this.boundSelectField in the constructor of a form component
 // to update select fields
 // pass in the name (used as placeholder), key for profile, and the options.
-export function boundDateField(key, label, error) {
-  const { updateProfile, profile } = this.props;
+export function boundDateField(keySet, label) {
+  const {
+    profile,
+    errors,
+    validationMessages,
+    updateProfile,
+  } = this.props;
+
   let onChange = date => {
-    let clone = Object.assign({}, profile);
+    let clone = _.deepClone(profile);
     // format as ISO-8601
-    clone[key] = date.format("YYYY-MM-DD");
-    updateProfile(clone);
+    _.set(clone, keySet, date.format("YYYY-MM-DD"));
+    updateProfile(clone)
   };
-  let newDate = profile[key] ? moment(profile[key], "YYYY-MM-DD") : null;
+  let newDate = _.get(profile, keySet, undefined) ? moment(_.get(profile, keySet), "YYYY-MM-DD") : null;
   return <div>
     <DatePicker
       selected={newDate}
       placeholderText={label}
       showYearDropdown
       onChange={onChange}
     />
-    <span className="validation-error-text">{error}</span>
+    <span className="validation-error-text">{_.get(errors, keySet, undefined)}</span>
   </div>;
 }
 
 
 // bind to this.saveAndContinue.bind(this, '/next/url')
-export function saveAndContinue(next) {
-  const { saveProfile, profile } = this.props;
+// pass an option callback if you need nested validation
+// (see EmploymentTab for an example)
+export function saveAndContinue(next, nestedValidationCallback) {
+  const {
+    saveProfile,
+    profile,
+    requiredFields,
+    validationMessages
+  } = this.props;
 
-  saveProfile(profile).then(() => {
+  let fields;
+  if ( _.isFunction(nestedValidationCallback) ) {
+    fields = nestedValidationCallback(profile, requiredFields);
+  } else {
+    fields = requiredFields;
+  }
+
+  saveProfile( profile, fields, validationMessages).then(() => {
     browserHistory.push(next)
   });
 }
+
+export function editProfileObjectArray (arrayName, blankEntry, formCallback) {
+  const { updateProfile, profile } = this.props;
+  if ( profile.hasOwnProperty(arrayName) && !_.isEmpty(profile[arrayName]) ) {
+    let editForms = profile[arrayName].map((v, i) => formCallback(i));
+    let addAnotherBlankEntry = () => {
+      let clone = Object.assign({}, profile);
+      clone[arrayName] = clone[arrayName].concat(Object.assign({}, blankEntry));
```

Comment:
just tried it out, seems like it's working just fine with those `Object.assign(...)` calls taken out.